### ISSUE TITLE
Inline expressions not working

### ISSUE THREAD

**Issue by zephyo** [YarnSpinnerTool/YarnSpinner-Unity#35]
**What is the current behavior?**

According to https://yarnspinner.dev/docs/syntax/, this line:
`Think: yarn var = {$popularity}.`

Should result in:
`Think: yarn var = 30.`

But instead, I get the following:
`Think: yarn var = {0}.`

**Please provide the steps to reproduce, and if possible a minimal demo of the problem**:

1. Have yarn file: 
```
title: Start
tags:
---
Think: yarn var = {$popularity}.
===
```
2. Be using develop branch.
3. Output to dialogue ui.
4. Add debug logs to the variable storage.
5. See the following logs (seems as if GetValue from variable storage is being called but the delivered line doesn't use the gotten value):
<img width="588" alt="logs" src="https://user-images.githubusercontent.com/22963510/92598854-681d4480-f277-11ea-9f1a-483090e81020.png">

Note: I notice there's a similar issue for localization, but I'm not localizing anything.

**What is the expected behavior?**

I expect the inline expressions to work as documented.

**Please tell us about your environment:**
  
  - Yarn Spinner Version: Most recent develop branch
  - Unity Version: 2020.1.4

**Other information**

**Comment by jonathaniscarroll** [comment]
I was finding this as well, realized the problem was i was feeding InMemoryVariableStorage.SetValue() a string for the variableName that did not have a $ in front of it. Instead of setting the variable (called $score), it was creating a new variable (called score) with the new value.

**Comment by zephyo** [comment]
Ah interesting... for me, the variable does have the $ prepended (see debug log screenshot - '$popularity')

**Comment by desplesda** [comment]
This is fixed in 284fd60accad852aaf6b34cdcbbd5cc9f7033954. Closing; thanks for filing this issue!
